lms/views/api/exceptions.py

```diff
@@ -103,10 +104,15 @@ def __init__(self, context, request):
         self.context = context
         self.request = request
 
+        # 400 is the default status for error responses.
+        # Some of the views below override this.
+        self.request.response.status_int = 400
```

Comment:
The `error_response()` helper method used to set `self.request.response.status_int` _and_ return the error response body dict. So it was doing two things. It would be weird if the new `ErrorResponseBody` dataclass would set `self.request.response.status_int`: the class just represents the the response _body_, it has nothing to do with the status code. So I just moved status code setting into the views.

(I could write a class to represent the whole error response, not just the body. Then views could just return an instance. But it would be more work because Pyramid's JSON renderer represents a class that represents only the response body, and we get a lot for free by using the JSON renderer.)

Reply:
By the looks of it if you return a `Response` object it skips the `json` renderer entirely:

 * https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/renderers.html#writing-view-callables-which-use-a-renderer
 
```python
from pyramid.response import Response
from pyramid.view import view_config

@view_config(renderer='json')
def view(request):
    return Response('OK') # json renderer avoided
```

So unless we wanted to replicate the whole shebang, this seems like a real pain to do. Also... why this seems fine to me.

Reply:
Yeah, that's how Pyramid works. So whenever a view is using a renderer (even, say, the Jinja2 template renderer) it can always just bypass that by returning a response directly. For example you can return an `HTTPRedirect` (since all the HTTP exceptions are `Request` subclasses).

I'm sure we could do some kind of `JSONResponse` class but I agree that working with the builtin JSON renderer seems to be going with the flow here, and individual views doing `request.response.status_int = <whatever>` is also very Pyramid